src/brsTypes/components/Group.ts

```diff
@@ -0,0 +1,29 @@
+import { RoSGNode } from "./RoSGNode";
+import { ValueKind } from "../BrsType";
+import { AAMember } from "./RoAssociativeArray";
+
+export class Group extends RoSGNode {
+    readonly kind = ValueKind.Object;
+
+    readonly builtInFields = [
+        { name: "visible", type: "boolean" },
+        { name: "opacity", type: "float" },
+        { name: "translation", type: "array" },
+        { name: "rotation", type: "float" },
+        { name: "scale", type: "array" },
+        { name: "scaleRotateCenter", type: "array" },
+        { name: "childRenderOrder", type: "string" },
+        { name: "inheritParentTransform", type: "boolean" },
+        { name: "inheritParentOpacity", type: "boolean" },
+        { name: "clippingRect", type: "array" },
+        { name: "renderPass", type: "integer" },
+        { name: "muteAudioGuide", type: "boolean" },
+        { name: "enableRenderTracking", type: "boolean" },
+        { name: "renderTracking", type: "string" },
+    ];
+
+    constructor(members: AAMember[] = []) {
```

Comment:
Any chance this could accept a `FieldModel[]` (as defined in `RoSGNode.ts`) instead of an `AAMember[]`?  When we eventually add some typing for the list of supported `type` values, it'll be handy to have those plumbed through everywhere already 😃

Reply:
In this case, I think the logic would need to change so that the child component doesn't need to explicitly call `this.registerFields` for its own fields, i.e. the constructor logic for `Group` (or better yet, `RoSGNode`) would need to look something like:
```
constructor(members: FieldModel[] = []) {
  super("RoSGNode");
  let allFields = [...this.builtInFields, ...members];
  this.registerFields(allFields);
}
```

Reply:
Or actually, I guess you could still have the child call `this.registerFields` for its own fields, instead of passing them to the parent. Not sure which is more desirable 🤷‍♂️

Reply:
True!  I think I like the explicit `this.registerFields()` call over passing things through the constructor — if only because you could make the same argument about `this.registerMethods()` and constructor calls would be really large after adding both of those 😅

Reply:
Also, second thought: don't we need `members` to have a value associated with them? Versus the `builtInFields`, which are all initialized to the default value. I assumed that when we had a component in Scenegraph like:
```
<Group foo="bar">
```
then we would be initializing it along the lines of
```
let members = [{ name: "foo", value: "bar" }];
let group = new Group(members);
```

And then the `builtInFields` are all of the interfaces that _can_ be defined on a certain node, initialized to default values, then overwritten by `members` if applicable.

Of course, that was just my assumption based on how the class is constructed, and I haven't had a chance to dive into the flow of parsing/initializing yet, so please correct me if that assumption is wrong 😅

Reply:
I _believe_ `members` would represent the _children_ of the component in question... but that's a good point to clarify. 😄 
```
<Group>
    <Rectangle />
    <String />
</Group>
```

Reply:
I'm also finding the inconsistency with the naming confusing. Fields and members are comparable? If it's the children we might want to be more explicit about that as well.